### PR TITLE
table limit range support

### PR DESCRIPTION
**Pull request by code2prog**
I added support for offset in table limit
sample usage:
`
$dump->setTableLimits([
                'table_name' => [10,200],
            ]);
`

I think this may also be useful for other users. There have been several requests over the years to add this functionality.

### CODE REVIEW

**Comment by back-2-95** [comment]
@code2prog Thanks for the PR! Could you add some links to old issues/PRs to get some background?

**Comment by code2prog** [comment]
@back-2-95 https://github.com/ifsnop/mysqldump-php/issues/175

File: src/Mysqldump.php [hunk 1]
```diff
@@ -1037,7 +1040,19 @@ public function getTableLimit(string $tableName)
             return false;
         }
 
-        return is_numeric($this->tableLimits[$tableName]) ? $this->tableLimits[$tableName] : false;
+        $v = false;
+        if (is_numeric($this->tableLimits[$tableName])) {
+            $v = is_numeric($this->tableLimits[$tableName]);
```
Comment (back-2-95) [inline src/Mysqldump.php:1045]: Should this `return $this->tableLimits[$tableName];` instead returning `true`?

Comment (code2prog) [reply src/Mysqldump.php:1045]: It is possible that you are right. I'll check it on Monday and change it.

Comment (code2prog) [reply src/Mysqldump.php]: @back-2-95 I changed it and added additional tests
https://github.com/druidfi/mysqldump-php/pull/16/commits/0196b1bc986dc20dd26acb476de4ff47cdf9d8fd

Comment (back-2-95) [reply src/Mysqldump.php]: Code looks valid, I'll run the tests

File: src/Mysqldump.php [hunk 2]
```diff
@@ -1037,7 +1040,21 @@ public function getTableLimit(string $tableName)
...
+        if (is_array($this->tableLimits[$tableName]) &&
+            count($this->tableLimits[$tableName]) === 2 &&
+            is_numeric(implode('', $this->tableLimits[$tableName])) &&
+            $this->tableLimits[$tableName][0] <= $this->tableLimits[$tableName][1]
+        ) {
+            $limit = implode(',', $this->tableLimits[$tableName]);
+        }
+
+        return $limit;
+
```
Comment (back-2-95) [inline src/Mysqldump.php:1057]: Please remove this empty line.

File: src/Mysqldump.php [hunk 3]
```diff
@@ -1037,7 +1040,21 @@ public function getTableLimit(string $tableName)
             return false;
         }
 
-        return is_numeric($this->tableLimits[$tableName]) ? $this->tableLimits[$tableName] : false;
+        $limit = false;
```
Comment (back-2-95) [inline src/Mysqldump.php:1043]: Add empty line after this.

**Comment by back-2-95** [comment]
Could you also update `README.md` section `## Table specific export limits` by adding limit to posts like `[0,100]` ? And perhaps some sentence about offset.

**Comment by code2prog** [comment]
I will add it tomorrow morning and improve the code style

File: src/Mysqldump.php (hunk 1 above)
Comment (code2prog) [reply src/Mysqldump.php]: @back-2-95 https://github.com/druidfi/mysqldump-php/pull/16/commits/aeca7c0791d003934fb2ee2329a63f68f10e8116

**Pull request merged**